test/built-ins/String/prototype/split/errors.js

```diff
@@ -0,0 +1,124 @@
+}
+
+// Start with input consisting of descriptors for values
+// that should each throw an error at the earliest possible step.
+var input = {
+  _this: { value: undefined, errorType: TypeError },
+  separator: badValueDescriptor("SeparatorGetSplitError", function(UniqueError) {
+    return new Proxy({}, {
+      get(target, prop) {
+        if (prop === Symbol.split) {
+          throw new UniqueError();
+        }
+        return target[prop];
+      }
+    });
+  }),
+  limit: badValueDescriptor("LimitToUint32Error", function (UniqueError) {
+    return { valueOf() { throw new UniqueError(); } };
+  })
+};
+
+// After veryifying the expected error, replace the responsible descriptor
+// with a new one that throws the next specified error, ultimately using
+// a descriptor corresponding to "good" input.
+assert.throws(input._this.errorType, invoker(input),
+    "Abrupt completions from RequireObjectCoercible(*this* value) are propagated.");
+input._this = badValueDescriptor("ReceiverToStringError", function (UniqueError) {
+  return { toString() { throw new UniqueError(); } };
+});
+
+// Proceed accordingly through each step that can throw an error.
+assert.throws(input.separator.errorType, invoker(input),
+    "Abrupt completions from GetMethod(_separator_, @@split) are propagated.");
+input.separator = badValueDescriptor("SeparatorCallSplitError", function(UniqueError) {
+  return Object.defineProperty({}, Symbol.split,
+      { value: function () { throw new UniqueError(); } });
+});
+
+assert.throws(input.separator.errorType, invoker(input),
+    "Abrupt completions from Call(_splitter_, _separator_, &laquo; _O_, _limit_ &raquo;)" +
+    " are propagated.");
+input.separator = {
+  value: Object.defineProperty({}, Symbol.split, { value: function () {} })
+};
+
+lastCallCount = callCount;
+invoker(input)();
+assert.sameValue(callCount, lastCallCount + 1,
+    "No steps are evaluated after Call(_splitter_, _separator_, &laquo; _O_, _limit_ &raquo;)");
+input.separator = badValueDescriptor("SeparatorToStringError", function (UniqueError) {
+  return Object.defineProperty(
+    { toString() { throw new UniqueError(); } },
+    Symbol.split,
+    { value: undefined });
+});
+
+assert.throws(input._this.errorType, invoker(input),
+    "Abrupt completions from ToString(_O_) are propagated.");
+input._this = { value: "" };
+
+assert.throws(input.limit.errorType, invoker(input),
+    "Abrupt completions from ToUint32(_limit_) are propagated.");
+input.limit = { value: 0 };
+
+assert.throws(input.separator.errorType, invoker(input),
+    "Abrupt completions from ToString(_separator_) are propagated.");
+input.separator = {
+  value: Object.defineProperty(
+    { toString() { return undefined; } },
+    Symbol.split,
+    { value: undefined })
+};
+
+// End with a verification of non-abrupt completion when invoking
+// the final (all "good") invoker.
+lastCallCount = callCount;
+invoker(input)();
+assert.sameValue(callCount, lastCallCount + 1);
```

Comment:
One bit before I have to put this away until next week: we're trying to break tests up into smaller chunks, not consolidate. I'd be more interested in seeing a new harness helper file, if you want to propose something like that? Either way, these all need to return to their original files.

Reply:
My position motivating this PR is that having such granularity and spreading tests over so many files invites coverage gaps—I'm explicitly advocating _for_ consolidation, specifically of all abrupt completions for an operation into one file.

Reply:
_It's not a big deal for a simple case like this PR, but I would slightly prefer detailed results_.

@gibson042 I understand this motivation and sympathize with it but there are some catches.

The flow for tests execution and error reporting depends on the way we handle assertions. It is ridiculously unfortunate that our assertions are designed to throw on failures rather than reporting or storing the failures and preventing code to continue executing. All of us here knows the advantages of an improved flow like we've had from QUnit. I miss it a lot in Test262.

That leads us for the case where an initial failure does prevent a detailed report, hence that's one big reason why we have multiple files.
 
My biggest take here is that I'd love to enhance the current system available for Test262 API, with something that allows a better grouping of assertions to preserve a consistent error reporting. It's important that we provide better tools to permit consolidation of tests capture the same groups of abrupt completions, etc. I believe we lack those in our API and there are different ways to tackle this. I believe we are in a good opportunity here as our bias somehow connect to QUnit and it might be a good path to discuss for what we might want to design.

Reply:
So is there a path forward here? Ordering is an observable and therefore testable part of the spec, but having O(N) files dedicated to it for _each_ N-parameter function seems completely unmaintainable.